anybox/recipe/odoo/vcs/git.py

```diff
@@ -414,19 +445,25 @@ def merge(self, revision):
                                    "or non git local directory %s" %
                                    self.target_dir)
             os.chdir(self.target_dir)
-            rtype, sha = self.query_remote_ref(self.url, revision)
-            if rtype is None and ishex(revision):
-                self.fetch_remote_sha(revision, checkout=False)
+            if ishex(revision) and self.has_commit(revision):
+                # if pinned, try to find on local first
                 cmd = ['git', 'merge', revision]
-            else:
-                cmd = ['git', 'pull', self.url, revision]
-            if self.git_version >= (1, 7, 10):
-                # --edit and --no-edit appear with Git 1.7.10
-                # see Documentation/RelNotes/1.7.10.txt of Git
-                # (https://git.kernel.org/cgit/git/git.git/tree)
-                cmd.insert(2, '--no-edit')
-
-            self.log_call(cmd)
+                self.log_call(self._no_edit(cmd))
+                return
+
+            rtype, sha = self.query_remote_ref(BUILDOUT_ORIGIN, revision)
```

Comment:
Please restore `self.url` instead of `BUILDOUT_ORIGIN`(from https://github.com/anybox/anybox.recipe.odoo/commit/c8948875402902c1ec7e7592d839f196b49888f0#diff-e3c884172ce5feb7095a934b88af6da2d4c4ccaaafc7b87bfab2b8167ad6e446L416-R417).

That also makes commit https://github.com/anybox/anybox.recipe.odoo/pull/104/commits/162c27715e redundant, so please leave it out of this PR, as it breaks merging by pinned hash commit.

You should end up with something like 
```
    def merge(self, revision):
        """Merge revision into current branch"""
        with working_directory_keeper:
            if not self.is_versioned(self.target_dir):
                raise RuntimeError("Cannot merge into non existent "
                                   "or non git local directory %s" %
                                   self.target_dir)
            os.chdir(self.target_dir)
            if ishex(revision) and self.has_commit(revision):
                # if pinned, try to find on local first                                                                                                                                                                                  
		cmd = ['git', 'merge', revision]
                self.log_call(self._no_edit(cmd))
                return

            rtype, sha = self.query_remote_ref(self.url, revision)
	    if rtype is None and ishex(revision):
                self.fetch_remote_sha(revision, checkout=False)
		cmd = ['git', 'merge', revision]
            else:
                cmd = ['git', 'pull', self.url, revision]

            self.log_call(self._no_edit(cmd))
```